clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -222,15 +236,43 @@ private EhcacheEntityResponse invokeInternal(TimeoutDuration timeLimit, EhcacheE
     }
   }
 
-  private InvokeFuture<EhcacheEntityResponse> internalInvokeAsync(EhcacheEntityMessage message, boolean track)
-        throws MessageCodecException {
-    return endpoint.beginInvoke().message(message).invoke();
+  private InvokeFuture<EhcacheEntityResponse> invokeInternal(InvocationBuilder<EhcacheEntityMessage, EhcacheEntityResponse> invocationBuilder, long timeout, EhcacheEntityMessage message, boolean track) throws TimeoutException {
+    boolean interrupted = Thread.interrupted();
+    try {
+      long end = System.nanoTime() + timeout;
+      while (true) {
+        try {
+          long nanos = end - System.nanoTime();
+          if (nanos > 0) {
+            return invocationBuilder.message(message).invokeWithTimeout(nanos, TimeUnit.NANOSECONDS);
+          } else {
+            throw new TimeoutException();
+          }
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
Since we are interrupted, I would prefer to go out right away. Returning null I guess